### ISSUE TITLE
Time and date field contents aren't validated

### ISSUE THREAD

**Issue by barbeau** [CUTR-at-USF/usf-mobullity#210]
From USF UCM:

> No validation on the time field, and no error message, so entering a non-numerical value, and submitting does nothing.

We should validate the time field value before accepting it.

![image](https://cloud.githubusercontent.com/assets/928045/13542699/935e0062-e232-11e5-9b14-46950677736f.png)

**Comment by jailby** [comment]
The block of code at [this location](https://github.com/CUTR-at-USF/usf-mobullity/blob/mobullityrebase/src/client/js/otp/widgets/tripoptions/TripOptionsWidget.js#L412) transform the time entered into the input field into the am/pm format. Thus if the user uses the 24 hours format it will automatically be changed to the 12 hours format.

However, it is possible to enter values greater than 24 (e.g., the value 33:00 is transformed to 9pm), this should return an error similar as if the user enter a non numerical character.

The input should match this regex: 
`((([0-9]|0[0-9]|1[0-9]|2[0-3]):[0-5][0-9]) | (([0-9]|0[0-9]|1[0-2]):[0-5][0-9])((a|p)m))`

If it is not matching this regex we could show a popup to let know the user.

**Comment by jailby** [comment]
Screenshot of the popup indicating the user that the time is not correct:

![image](https://cloud.githubusercontent.com/assets/1370401/14321566/b9779622-fbe7-11e5-95b3-6a7db7f6203b.png)

This popup is trigger when the time value is changed. Which means that the user can discard the information in the popup and still try to plan a trip.

Maybe for consistency with the popup that appear when the start and end location field are not populated, we should only make the popup appear only when the user click "Plan Your Trip".

I also would like to find a regex that has been use and somewhat agreed on for time. Since I made this one it is easy to have a small mistake/exception left out.

**Comment by barbeau** [comment]
Looks good!  I'd make a slight change to the message:

> Please enter a valid time in 12 hr or 24 hr format.

I would trigger the validation when the time value is changed, but when detecting the invalid time, I'd change the time value back to the previous time (which I believe by default is "now").  So, the user will be forced to always plan a trip with a valid time.

**Comment by jailby** [comment]
I changed the text.

I also made this popup triggered if the user tries to plan a trip with an invalid time.

So right now it works as follow:
If a user writes an invalid time the popup appear (which is basically a warning).
The user can discard the popup and still hit the button "Plan Your Trip". If the time is still invalid the popup appear again and the trip is not planned.

I think this solution offer more usability for the following reason:
I think that the most common invalid time would be a mistake (e.g.: a typo).

For example, if someone wants to plan a trip for when he/she get off from work at 5pm and writes "5ppm" this is an invalid time and with the solution of putting back the previous time it would force the user to re-write time when he just had to correct his typo.

I don't think it is a big deal but its an annoyance we can avoid.

I also don't think that the value entered by a user is never stored. So we would have to always store the value somewhere when the user change it to be able to revert it back.
